### PR TITLE
Add SW_FastEdit code

### PR DESCRIPTION
**Pull request by matt3o**
See [this discussion](https://github.com/Project-MONAI/MONAI/discussions/7580#discussioncomment-8975969) for details, also @diazandr3s 

[Paper link](https://arxiv.org/pdf/2311.14482.pdf)
Link to the [full code base](https://github.com/matt3o/AutoPET2-Submission).

Current limitations:
- all parameter are currently tuned for AutoPET
- currently inference only, I'll see if I can get training to work
- Pretrained weights only available for AutoPET, can be found here: https://bwsyncandshare.kit.edu/s/Yky4x6PQbtxLj2H , rename it to pretrained_sw_fastedit.pt and put them into the folder radiology/model/. This model was pretrained on tumor-only AutoPET volumes.
(where do can upload the weights?)
- main code is not integrated into MONAI, thus some additional transforms have been added

### CODE REVIEW

**Review by SachidanandAlle — APPROVED** [review]

**Comment by SachidanandAlle** [comment]
Can you fix the precommit checks.. you can run locally to identify the issues..

Please check the contrib guidelines to know the command to run the same locally

**Comment by matt3o** [comment]
Sure! The code also still contains some debug code which I will remove as well.

**Comment by matt3o** [comment]
@SachidanandAlle  Should work with this commit. If you want to try it yourself, use the following weights file https://drive.google.com/file/d/1p59s-ehEyp39vlqULAd1eB8gV4hkmxR0/view?usp=drive_link
(I changed the code to use the newer, AutoPET 3rd place ScaleIntensityRangePercentilesd, which helps to make the results more robust)

Please tell me if there is anything else you need.

**Comment by SachidanandAlle** [comment]
Assigning to @diazandr3s to verify and merge

**Comment by diazandr3s** [comment]
Hi @matt3o,

Many thanks for your contribution and sorry I just came back to you.

Just to confirm, this model doesn't have the training class, right? can you please comment more on how you trained the model?

I've uploaded the model to the MONAI Label repository - now users should be able to run inference as the other radiology models. Here is the link to the model for your reference: https://github.com/Project-MONAI/MONAILabel/releases/download/pretrained/radiology_segmentation_sw_fastedit_pet.pt

I've also made a video shwoing how this model works using a sample PET I have:

https://github.com/Project-MONAI/MONAILabel/assets/11991079/97787b8d-54d9-4315-b6c7-8620d4ea533e

Many thanks again!

**Comment by matt3o** [comment]
Very nice, thanks for the integration :) Sadly I did not find the time to implement the training part. Please ping if someone is asking for is specifically and I'll see if I can get it to work. I think the greatest benefit of the model is specifically the non-interactive pretraining part - the rest is only slightly different from DeepEdit.

About the training: 400 epochs of non-interactive pretraining on all AutoPET samples each epoch, so still with all three channels, but two of them are just zeroed out. Then 200 epochs of interactive training using the DeepEdit interactive sampling of clicks, but in very revamped manner - a lot faster and with some improvements. 
Some of the details and data can be found in this paper: https://arxiv.org/abs/2311.14482
What you are seeing is the model on the bottom right, which had the best initial segmentation while still improving a lot with added clicks.
image
